pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -61,7 +61,9 @@ type objectCacheItem struct {
 	lastAccessTime time.Time
 	stopped        bool
 	immutable      bool
-	stopCh         chan struct{}
+	parentCtx      context.Context
+	ctx            context.Context
```

Comment:
Nit: I can see the need to store `parentCtx` for `restartReflectorIfNeeded`, but `ctx` I would try to pass as function parameter.

Reply:
Yeah, I think this is a good idea.